common/src/main/java/com/microsoft/identity/common/internal/net/HttpResponse.java

```diff
@@ -102,4 +102,17 @@ public String toString() {
                 '}';
     }
     //CHECKSTYLE:ON
+}
+
+private Map<String, List<String>> filterNullKeyHeaders(@NonNull final Map<String, List<String>> headerFields)    
+{        
+    Map<String, List<String>> filteredHeaders = new HashMap<String, List<String>>();        
+    for(String key: headerFields.keySet())        
+    {           
+        if(key!= null)            
+        {                
+            filteredHeaders.put(key,headerFields.get(key));            
+        }
+    }
+    return filteredHeaders;
```

Comment:
Why are we doing this in MSAL common?  This seems to be a problem specific to the C++ applications, and should be fixed in the adapter layer there.

Reply:
Thing is, this is actually fixed in MSAL-Cpp now.  So I'm not sure we need to do this here.  If we were going to take on moving this to a new field, I'd create one in HttpResponse called mStatusLine and put it there.  The question is - do we need to do this at all?